src/output.rs

```diff
@@ -38,15 +45,43 @@ pub fn print_entry<W: Write>(stdout: &mut W, entry: &Path, config: &Config) {
     }
 }
 
+// Display a trailing slash if the path is a directory and the config option is enabled.
+// If the path_separator option is set, display that instead.
+// The trailing slash will not be colored.
+#[inline]
+fn print_trailing_slash<W: Write>(
+    stdout: &mut W,
+    entry: &DirEntry,
+    config: &Config,
+    style: Option<&Style>,
+) -> io::Result<()> {
+    if entry.file_type().map_or(false, |ft| ft.is_dir()) {
+        let separator = config
+            .path_separator
+            .as_ref()
+            .unwrap_or(&MAIN_SEPARATOR_STR);
```

Comment:
I believe this will have to perform an atomic operation every time we print an entry. It would probably be more performant to store the actual path seperator once in config rather than in a `Lazy`. 

Although looking at it some more, it might actually be better to change `path_separator` to be an `Option<char>` instead of an `Option<String>`, since we require it to be a single byte anyway.

Reply:
I'll look more into changing this to use `char` in a seperate branch.

Reply:
hmm, actually it is only required to be a single byte on windows. Although I don't know if there is much of a use case for it to be more than one character.

Reply:
yeah, I don't think it's worth changing this to use a char. 

However, I do think having a `config.actual_path_separator` that is either the wrapped value of `config.path_separator` or the default path separator.